src/main/java/com/getindata/connectors/http/internal/OIDCAuthHeaderValuePreprocessor.java

```diff
@@ -0,0 +1,50 @@
+package com.getindata.connectors.http.internal;
+
+import java.net.http.HttpClient;
+import java.time.Duration;
+import java.util.Optional;
+
+import lombok.extern.slf4j.Slf4j;
+
+import com.getindata.connectors.http.internal.auth.OidcAccessTokenManager;
+
+/**
+ * Header processor for HTTP OIDC Authentication mechanism.
+ */
+@Slf4j
+public class OIDCAuthHeaderValuePreprocessor implements HeaderValuePreprocessor {
+
+
+    private final String oidcAuthURL;
+    private final String oidcTokenRequest;
+    private Duration oidcExpiryReduction = Duration.ofSeconds(1);
+    /**
+     * Add the access token to the request using OidcAuth authenticate method that
+     * gives us a valid access token.
+     * @param oidcAuthURL  OIDC token endpoint
+     * @param oidcTokenRequest  OIDC Token Request
+     * @param oidcExpiryReduction OIDC token expiry reduction
+     */
+
+    public OIDCAuthHeaderValuePreprocessor(Optional<String> oidcAuthURL,
+                                           Optional<String> oidcTokenRequest,
+                                           Optional<Duration> oidcExpiryReduction) {
+        this.oidcAuthURL = oidcAuthURL.get();
```

Comment:
nit: Do not pass Optional<String> if the parameter is required (`.get() is called`). Pass `String` instead.